src/lib/services/setting-service.ts

```diff
@@ -2,30 +2,67 @@ import { IUnleashConfig } from '../types/option';
 import { IUnleashStores } from '../types/stores';
 import { Logger } from '../logger';
 import { ISettingStore } from '../types/stores/settings-store';
+import { IEventStore } from '../types/stores/event-store';
+import {
+    SettingCreatedEvent,
+    SettingDeletedEvent,
+    SettingUpdatedEvent,
+} from '../types/events';
 
 export default class SettingService {
     private logger: Logger;
 
     private settingStore: ISettingStore;
 
+    private eventStore: IEventStore;
+
     constructor(
-        { settingStore }: Pick<IUnleashStores, 'settingStore'>,
+        {
+            settingStore,
+            eventStore,
+        }: Pick<IUnleashStores, 'settingStore' | 'eventStore'>,
         { getLogger }: Pick<IUnleashConfig, 'getLogger'>,
     ) {
         this.logger = getLogger('services/setting-service.ts');
         this.settingStore = settingStore;
+        this.eventStore = eventStore;
     }
 
     async get<T>(id: string): Promise<T> {
         return this.settingStore.get(id);
     }
 
-    async insert(id: string, value: object): Promise<void> {
-        return this.settingStore.insert(id, value);
+    async insert(id: string, value: object, createdBy: string): Promise<void> {
+        const exists = await this.settingStore.exists(id);
+        if (exists) {
+            await this.settingStore.insert(id, value);
```

Comment:
Is this meant to be an insert? I see the store has a `updateRow` method. If the insert is intentional, it might be worthwhile to move the insert out of the if statement, mostly because my eyes are telling me that this is intentional to have the insert in the if body and that the reason both branches are using the insert is because it's a typo

Reply:
Good point. The insert behind the scenes also checks if it exists. I've changed to call updateRow